core/store/src/archive/cold_storage.rs

```diff
@@ -97,7 +98,7 @@ pub fn update_cold_db(
             if !col.is_cold() {
                 return false;
             }
-            if col == &DBCol::StateShardUIdMapping && !is_last_block_in_epoch {
+            if col == &DBCol::StateShardUIdMapping {
```

Comment:
should we just return false for col.is_cold() in case of `StateShardUIdMapping`? (not sure if it would impact anything else, or if there are some additional clean-ups that can be done)